shared/reducers/chat2.js

```diff
@@ -735,15 +750,46 @@ const rootReducer = (
         }
       )
 
+      let containsLatestMessageMap = state.containsLatestMessageMap.withMutations(map => {
+        Object.keys(convoToMessages).forEach(cid => {
+          const conversationIDKey = Types.stringToConversationIDKey(cid)
+          if (map.get(conversationIDKey, false)) {
+            return
+          }
+          const meta = state.metaMap.get(conversationIDKey, null)
+          const ordinals = messageOrdinals.get(conversationIDKey, I.OrderedSet()).toArray()
+          let maxMsgID = 0
+          for (let i = ordinals.length - 1; i >= 0; i--) {
+            const ordinal = ordinals[i]
+            const message = messageMap.getIn([conversationIDKey, ordinal])
```

Comment:
might want to pull out messageMap.get(conversationIDKey) outside the loop so we don't have to do this map lookup each time in here. likely doesn't matter but doesn't really add anything to not